models/quality_measures/intermediate/nqf2372_breast_cancer_screening/quality_measures__int_nqf2372_exclude_mastectomy.sql

```diff
@@ -169,9 +171,102 @@ with denominator as (
 
 )
 
+/*
+    Women who had a bilateral mastectomy or who have a history of a bilateral
+    mastectomy
+*/
+, bilateral_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+          'Bilateral Mastectomy'
+        , 'History of bilateral mastectomy'
+    )
+
+)
+
+, right_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+          'Status Post Right Mastectomy'
+        , 'Unilateral Mastectomy Right'
+    )
+
+)
+
+, left_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+          'Status Post Left Mastectomy'
+        , 'Unilateral Mastectomy Left'
+    )
+
+)
+
+, unspecified_mastectomy as (
+
+    select
+          patient_id
+        , exclusion_date
+        , exclusion_reason
+    from all_mastectomy
+    where exclusion_reason in (
+        'Unilateral Mastectomy, Unspecified Laterality'
+    )
+
+)
+
+/*
+    Women for whom there is evidence of a right AND a left unilateral mastectomy
+    or unspecific mastectomies on different dates
+*/
+, unilateral_mastectomy as (
+
+    select
+          right_mastectomy.patient_id
+        , right_mastectomy.exclusion_date
+        , right_mastectomy.exclusion_reason
+    from right_mastectomy
+         inner join left_mastectomy
+            on right_mastectomy.patient_id = left_mastectomy.patient_id
+
+    union all
+
+    select
+          unspecified_mastectomy.patient_id
+        , unspecified_mastectomy.exclusion_date
+        , unspecified_mastectomy.exclusion_reason
+    from unspecified_mastectomy
+         inner join unspecified_mastectomy as self_join
+            on unspecified_mastectomy.patient_id = self_join.patient_id
+            and unspecified_mastectomy.exclusion_date <> self_join.exclusion_date
```

Comment:
This is the best way I came up with to ensure we are applying the same logic for unilateral mastectomies when it's unspecified. This ensures they took place on different dates.